### ISSUE TITLE
CI for signed APK fails

### ISSUE THREAD

**Issue by eznix86** [dllewellyn/crypto_currency_wallet#1]
```
 Set up job3s
  Run actions/checkout@v12s
  Run actions/setup-java@v17s
  Run subosito/flutter-action@v1.1.11m 8s
  Run actions/upload-artifact@v11s
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.163.1/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Artifact.PublishArtifact, Runner.Plugins"'.
Run actions/upload-artifact@v1
##[error]Path does not exist /home/runner/work/crypto_currency_wallet/crypto_currency_wallet/**
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.163.1/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Artifact.PublishArtifact, Runner.Plugins"'.
  Complete job
```

**Comment by paaKways** [comment]
use v2 for upload-artifact instead of v1

**Comment by paaKways** [comment]
@dllewellyn Which brings this to my problem. The code sample from the blog post [here](https://medium.com/@danieln.llewellyn/flutter-github-actions-for-a-signed-apk-fcdf9878f660) did not work for me. 

I followed the steps exactly. It seems to me something fails on this line:
'echo $KEY_JKS > key.jks && flutter pub get && flutter test && flutter build apk --release'

I was able to do a few builds just executing:
```
'echo $KEY_JKS > key.jks
```
 and download the zipped folder of the project and in all instances I could not find the key.jks file even though it should have been in the artifact.

Also, I noticed in the manual keystore generation process you placed the release keystore (JKS file) in `android/key.jks` but in the command you echo to `/key.jks`

Wouldn't that cause an error when flutter tries to build release APK?

**Comment by acanoe** [comment]
TL;DR: consult [this file](https://github.com/acanoe/binoculars/blob/d6a6091a1f82c26cc9d264ec193d421d83ab03e6/.github/workflows/main.yml).

I have found the solution to this, starting with the base64 encode step, he wrote:

> On a mac do the following
> `openssl base64 -in key.jks`

I substituted that with `base64 key.jks` on my ubuntu machine.

Then, he put `echo $KEY_JKS > key.jks && flutter pub get && flutter test && flutter build apk --release`, what was causing the error was the `echo $KEY_JKS > key.jks` part, not only the resulting file (`key.jks`) will not be found by the `build.gradle` (he put `storeFile file("../key.jks")` which translates to `[project root]/android/key.jks`, but the command will put the file in `[project root]/key.jks`), but also the resulting `key.jks` will not work because he forgot to decode the base64 string back.

To make it work, change that line into `echo "$KEY_JKS" | base64 -d > android/key.jks && flutter pub get && flutter build apk --split-per-abi`.
